ResearchKit/Common/ORKFormStepViewController.m

```diff
@@ -609,13 +610,29 @@ - (BOOL)allNonOptionalFormItemsHaveAnswers {
 }
 
 - (BOOL)continueButtonEnabled {
-    return ([self numberOfAnsweredFormItems] > 0
-            && [self allAnsweredFormItemsAreValid]
-            && [self allNonOptionalFormItemsHaveAnswers]);
+    BOOL enabled = ([self numberOfAnsweredFormItems] > 0
+                    && [self allAnsweredFormItemsAreValid]
+                    && [self allNonOptionalFormItemsHaveAnswers]);
+    if (self.isBeingReviewed) {
+        return enabled && ![self.savedAnswers isEqualToDictionary:self.originalAnswers];
+    } else {
+        return enabled;
+    }
```

Comment:
Lets aim for a single return.
if (self.isBeingReview) {
     enabled = enabled && ![self.savedAnswers ...
}
return enabled